docs/src/advanced/statistics.md

```diff
@@ -133,34 +69,110 @@ assert stats[0]["mean"] == 1.125  # (1 * 0.5 + 2 * 0.0 + 3 * 1.0 + 4 * 0.25) / 4
 assert stats[0]["count"] == 1.75  # (0.5 + 0 + 1 + 0.25) sum of the coverage array
 ```
 
-When using with a `feature`, your code might look something like:
+#### Adjusting geometry `align_bounds_with_dataset=True`
 
-```python
-from rio_tiler.io import Reader
-from rio_tiler.constants import WGS84_CRS
+In rio-tiler `6.3,0` a new option has been introduced to reduce artifacts and produce more precise zonal statistics. This option is available in the low-level `reader.part()` method used in rio-tiler reader's `.feature()` and `.part()` methods.
 
-with Reader(path) as src:
-    # First get the array for the feature
-    data = src_dst.feature(
+```python
+with Reader("cog.tif") as src:
+    data = src.feature(
         shape,
         shape_crs=WGS84_CRS,
+        align_bounds_with_dataset=True,
     )
 
-    # Get the coverage % array, using ImageData.get_coverage_array method
     coverage_array = data.get_coverage_array(
-        shape, shape_crs=WGS84_CRS
+        shape,
+        shape_crs=WGS84_CRS,
     )
 
-    # Get statistics (ImageData.statistics is calling `rio_tiler.utils.get_array_statistics`)
     stats = data.statistics(coverage=coverage_array)
 ```
 
-!!! warnings
-    The coverage weights will only have influence on specific statistics:
+When passing `align_bounds_with_dataset=True` to the  `reader.part()` method (forwared from `.feature` or `.part` reader's method), rio-tiler will adjust the input geometry bounds to math the input dataset resolution/transform and avoid unnecessary resampling.
```

Comment:
When passing `align_bounds_with_dataset=True` to the `reader.part()` method (forwarded from `.feature` or `.part` reader methods), rio-tiler will adjust the input geometry bounds to match the input dataset resolution/transform and avoid unnecessary resampling.